### PR TITLE
fix(lynx): support runtime-loaded bundle lifecycles

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
Non-eager Module Federation exposed three independent Lynx Web lifecycle defects:

- decoded bundles executed with the worker, blob, or host URL instead of their fetched source, so Rspack `publicPath: "auto"` resolved later chunks from the wrong directory;
- Web Core had no public, retryable API that could fetch, decode, execute, and return a lazy bundle in either realm;
- `LynxTemplatePlugin` treated Module Federation's remote-only async chunk groups as emitted Lynx lazy bundles, producing empty encodes or invented `lynx_aci` URLs.

This PR fixes those boundaries without eager shares or private NativeApp arguments. External bundles retain their exact fetched URL; component chunks receive a virtual child URL so Rspeedy's automatic lazy-bundle directory correction remains valid. Background execution uses a source-aware proxy while preserving native Worker method receivers, and main-thread execution exposes the logical source through `document.currentScript`. Binary and JSON decoders both retain external-bundle mode, empty lazy main-thread chunks remain valid no-ops, and the new public `lynx.loadLazyBundle` path deduplicates concurrent work while evicting failed loads for retry. Lazy and external load caches remain independent.

The template plugin now evaluates JavaScript-emission eligibility per original async chunk group. Assetless remote placeholders retain an explicit tombstone even when another group shares the same `webpackChunkName`, while real local chunks continue to emit and map normally.

The regressions use real Rspack Module Federation and real Web Core external bundles. The downstream reference implementation in [module-federation/core#4907](https://github.com/module-federation/core/pull/4907) proves static remote imports, dynamic remote imports, runtime `loadRemote()`, async startup, nested lazy chunks, standalone Catalog, and federated host behavior across native/iOS/Web without `getPublicPath`, eager shares, private NativeApp arguments, or private lazy-loader imports.

Native FetchBundle background sections now select the executable lazy chunk instead of the generated app-service dispatcher. Web FetchBundle keeps its real app-service entry and CSS section behavior unchanged; the regression is covered in the template-plugin suite and in the downstream decoded native artifacts.

React now exposes its existing loader through `@lynx-js/react/experimental/lazy/load` so startup integrations can register it before application entry execution. The module is safe to import outside a Lynx runtime, and absent Web realm constants default to the JavaScript background path. The downstream adapter uses this public export before federation `asyncStartup`; it does not duplicate the loader or rely on a private file layout.

Validation:

- `corepack enable && pnpm install --frozen-lockfile`
- `pnpm turbo build` (70/70 tasks after merging current `main`)
- `pnpm dprint check`
- `pnpm biome check`
- `pnpm --filter @lynx-js/web-core test` (23 files, 374 tests)
- `pnpm --filter @lynx-js/template-webpack-plugin test` (11 files, 104 tests, including the Windows fixture-path regression)
- `pnpm exec tsc --noEmit --project packages/web-platform/web-core/tsconfig.json`
- `pnpm exec tsc --build packages/webpack/template-webpack-plugin/tsconfig.build.json --force`
- `pnpm --filter @lynx-js/web-core build`
- `pnpm --filter @lynx-js/web-core-e2e run build:external-libs`
- `pnpm --filter @lynx-js/web-core-e2e exec rspeedy build --config=tests/reactlynx/external-bundle.config.ts`
- `pnpm --filter @lynx-js/web-core-e2e exec playwright test tests/reactlynx.spec.ts --project=chromium --grep external-bundle` (1 test)
- `pnpm exec dprint check` on all changed source, test, fixture, and changeset files
- `pnpm exec changeset status`
- Reproduced the pkg.pr.new shallow merge-base failure from Actions run 29891295094 in a fresh depth-1 checkout; unshallowing the pinned PR merge SHA restores the required ancestry.

- Rechecked after merging upstream main with the [Rspack layers preview at `843bab005b`](https://pkg.pr.new/web-infra-dev/rspack/@rspack/core@843bab005b2413345eb6efc3aaaba2a398fca908): normal frozen install, full build (74/74 tasks), template plugin (117 tests), Web Core (462 tests), external-bundle configuration (38 tests), and externals loading (17 tests) passed. The integration uses upstream shared encoding helpers while retaining external-bundle flags, remote-chunk handling, and lifecycle behavior. Exact preview pins remain; no Cargo configuration differs from upstream main.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).
- [x] Changeset added, and when a BREAKING CHANGE occurs, it needs to be clearly marked (or not required).

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed automatic public-path resolution for externally loaded bundles so lazy asset URLs resolve relative to the bundle URL.
  * Improved script execution mapping by preserving original execution source URLs/receiver context for external and lazy bundles, treating empty lazy main-thread chunks as no-ops, and excluding assetless async chunk groups from lazy-bundle URL generation.
* **New Features**
  * Added `loadLazyBundle` for fetching, decoding, and executing lazy bundles (with retryable behavior) and returning module exports.
* **Tests**
  * Expanded e2e and unit coverage for publicPath/sourceURL preservation, lazy/external concurrency, retries, and chunk/assetless edge cases.

Latest validation adds the full 94-file React snapshot/worklet/guardrail suite (100% coverage), the focused React lazy-bundle and export guardrails (48/48), a React package build, and Web Core decoder coverage for callable lazy code with unscoped component CSS. The downstream core reference consumes pinned pkg.pr.new Lynx packages while these changes are unreleased, and now validates against the same Rspack layers preview. Its [exact-head matrix at `946b6a55d`](https://github.com/module-federation/core/actions/runs/29994545755) is green across native artifact, Web, real iOS Simulator, Metro Android/iOS, Node, and all other affected E2Es.

### CODE REVIEW

**Comment by PupilTong** [comment]
I think using runtime information to treat a bundle in `lazyComponent`/`external bundle` way is not good enough. Could u please think about updating the web encoder and adding a flag to label if a bundle file is an "external bundle" or a "lazy bundle" ?
for example add a `isExternalBundle` flag to config?
@ScriptedAlchemy 

Thank u so much. I will learn about the behavior of the external bundle in native lynx and get back to u soon.

**Comment by ScriptedAlchemy** [comment]
@PupilTong I updated the producer and encoder paths so external and lazy bundle modes are encoded explicitly (`isExternalBundle` and `isLazy`), with runtime detection retained only as a compatibility fallback. I also addressed the subsequent automated review findings and rebased the PR onto current `main`. When you have a chance, is there anything else you would like me to adjust or consider in the implementation?

File: packages/web-platform/web-core/ts/types/ExternalBundle.ts [hunk 1]
```diff
@@ -18,6 +18,11 @@ export interface ExternalBundleResponse {
   errorMsg: string;
 }
 
+export interface FetchBundleOptions {
+  /** Decode the bundle as a lazy component instead of an external container. */
+  isLazyBundle?: boolean;
+}
```
[the remark this answers is not in the document]
Comment (ScriptedAlchemy) [reply packages/web-platform/web-core/ts/types/ExternalBundle.ts:25]: Addressed in ba5e9fe31: `FetchBundleOptions` now includes `[key: string]: unknown`, preserving arbitrary existing option bags while documenting `isLazyBundle`.
